RHDEVDOCS-3556 - RN Z-Stream Updates (#43066)
openshift/openshift-docs: modules/cluster-logging-release-notes-5.3.z.adoc

//Z-stream Release Notes by Version

[id="cluster-logging-release-notes-5-3-4"]
= OpenShift Logging 5.3.4
This release includes link:https://access.redhat.com/errata/RHBA-2022:0411[RHBA-2022:0411 OpenShift Logging Bug Fix Release 5.3.4]

[id="openshift-logging-5-3-4-bug-fixes"]
== Bug fixes

> * Before this update, changes to the metrics dashboards had not yet been deployed because the cluster-logging-operator did not correctly compare existing and desired configmaps containing the dashboard.  This update fixes the logic by adding unique hash value to the object labels. (link:https://issues.redhat.com/browse/LOG-2066[LOG-2066])

Reviewer: ```suggestion
* Before this update, changes to the metrics dashboards had not yet been deployed because the `cluster-logging-operator` did not correctly compare existing and desired config maps that contained the dashboard.  This update fixes the logic by adding a unique hash value to the object labels. (link:https://issues.redhat.com/browse/LOG-2066[LOG-2066])
```

* Before this update, Elasticsearch pods failed to start after updating with FIPS enabled. With this update, Elasticsearch pods start successfully. (link:https://issues.redhat.com/browse/LOG-1974[LOG-1974])

> * Before this update, elasticsearch generated the error "Unable to create PersistentVolumeClaim due to forbidden: exceeded quota: infra-storage-quota." if the PVC already existed. With this update, elasticsearch checks for existing PVCs, resolving the issue. (link:https://issues.redhat.com/browse/LOG-2127[LOG-2127])

Reviewer 2: Same two comments (parentheses and lower-case 'elasticsearch') for modules/cluster-logging-release-notes-5.2.z.adoc also apply to this line

Reviewer: s/elasticsearch/Elasticsearch/?

Reviewer: Migrate updates from previous release notes to similar ones, like this one, further down.

== CVEs
[id="openshift-logging-5-3-4-CVEs"]
.Click to expand CVEs
[%collapsible]
====

> * link:https://access.redhat.com/security/cve/CVE-2021-3521[CVE-2021-3521]

Reviewer 2: I can access only the first/top issue from this list, maybe because I don't have OpenShift subscription? I get **Bad Request** on the rest of this list.

[id="openshift-logging-5-3-2-bug-fixes"]
== Bug fixes
* Before this update, Elasticsearch rejected logs from the Event Router due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, previous indices might cause warnings or errors within Kibana. The `kubernetes.event.metadata.resourceVersion` field causes errors until existing indices are removed or reindexed. If this field is not used in Kibana, you can ignore the error messages. If you have a retention policy that deletes old indices, the policy eventually removes the old indices and stops the error messages. Otherwise, manually reindex to stop the error messages. (link:https://issues.redhat.com/browse/LOG-2087[LOG-2087])

* Before this update, the OpenShift Logging Dashboard displayed the wrong pod namespace in the table that displays top producing and collected containers over the last 24 hours. With this update, the OpenShift Logging Dashboard displays the correct pod namespace. (link:https://issues.redhat.com/browse/LOG-2051[LOG-2051])

> * Before this update, any situation with Elasticsearch OutputDefault defined for cluster logforwarding and Elasticsearch outputs did not have structured keys, the configuration was replacing the output secret with the default secret used to communicate to the default log store. With this update, the defined output secret is correctly used. (link:https://issues.redhat.com/browse/LOG-2046[LOG-2046])

Reviewer: ```suggestion
* Before this update, if `outputDefaults.elasticsearch.structuredTypeKey` in the `ClusterLogForwarder` custom resource (CR) instance did not have a structured key, the CR replaced the output secret with the default secret used to communicate to the default log store. With this update, the defined output secret is correctly used. (link:https://issues.redhat.com/browse/LOG-2046[LOG-2046])
```

Reviewer: This is my best guess at what this release note might mean. I recommend having an SME/QE review and approve it. Thanks.
